web-app/HeimScripts/heatmap/run.R

```diff
@@ -393,3 +507,43 @@ addClusteringOutput <- function(jsn, measurements) {
   )
   return(jsn)
 }
+
+### Limma part. It will be moved into a separate
+# module when we get R scripts sourcing to work here.
+
+getDEgenes <- function(df) {
+  measurements    <- getMeasurements(df)
+  enoughSamples   <- ncol(measurements) > 3 && ncol(measurements) - getSubset1Length(measurements) > 1
+  if (!enoughSamples){
+    stop("Not enough samples to find differentially expressed genes.")
+  }
+  design          <- getDesign(measurements)
+  contrast.matrix <- makeContrasts( S1-S2, levels = design )
+  fit             <- lmFit(measurements, design)
+  fit             <- contrasts.fit(fit, contrast.matrix)
+  fit             <- eBayes(fit)
+  contr           <- 1  #  We need a vector, not a df, so we'll do [, contr] on all stats
+  top.fit         <- data.frame (
+    logfold = fit$coefficients[, contr],
```

Comment:
For the calculation of log fold change: shouldn't this be done on "raw" values and then the log over that?